src/lustre/lustreTypeChecker.ml

```diff
@@ -41,13 +41,42 @@ type tc_type  = LA.lustre_type
               
 (** String of the type to display in type errors *)
 let string_of_tc_type: tc_type -> string = fun t -> Lib.string_of_t LA.pp_print_lustre_type t
+
+
+(** Pretty print type synonyms*)
+let pp_print_type_syn ppf = fun (i, ty) -> 
+  Format.fprintf ppf "(%a:=%a)" LA.pp_print_ident i LA.pp_print_lustre_type ty
+
+(** Pretty print type bindings*)
+let pp_print_type_binding ppf = fun (i, ty) -> 
+  Format.fprintf ppf "(%a:%a)" LA.pp_print_ident i LA.pp_print_lustre_type ty
+
+(** Pretty print value bindings (used for constants)*)
+let pp_print_val_binding ppf = fun (i, (v, ty)) ->
+  Format.fprintf ppf "(%a:%a :-> %a)" LA.pp_print_ident i LA.pp_print_lustre_type ty LA.pp_print_expr v
+
                                                   
 (** Map for types with identifiers as keys *)
-module IMap = Map.Make(struct
-                  type t = LA.ident
-                  let compare i1 i2 = Stdlib.compare i1 i2
-                end)
-
+module IMap = struct
+  (** everything that [Stdlib.Map] gives us  *)
+  include Map.Make(struct
+              type t = LA.ident
+              let compare i1 i2 = Stdlib.compare i1 i2
+            end)
+  (** Pretty print type context *)
+  let pp_print_ty_syns: Format.formatter -> 'a t -> unit
+    = fun ppf m -> Lib.pp_print_list (pp_print_type_syn) ", " ppf (bindings m)
+
+  (** Pretty print type context *)
+  let pp_print_tymap: Format.formatter -> 'a t -> unit
+    = fun ppf m -> Lib.pp_print_list (pp_print_type_binding) ", " ppf (bindings m)
+
+  (** Pretty print value store *)
+  let pp_print_vstore: Format.formatter -> 'a t -> unit
+    = fun ppf m ->  Lib.pp_print_list (pp_print_val_binding) ", "  ppf (bindings m)
```

Comment:
Keep these functions outside the IMap module.

Reply:
Fixed.